core/controllers/collection_editor_test.py

```diff
@@ -0,0 +1,92 @@
+
+"""Tests for the collection editor page."""
+
+__author__ = 'Abraham Mgowano'
+
+
+import os
+import StringIO
+import zipfile
+
+from core.controllers import collection_editor
+from core.domain import config_services
+from core.domain import collection_domain
+from core.domain import collection_services
+from core.domain import rights_manager
+from core.domain import rule_domain
+from core.tests import test_utils
+import feconf
+
+
+class BaseCollectionEditorControllerTest(test_utils.GenericTestBase):
+
+    CAN_EDIT_STR = 'GLOBALS.can_edit = JSON.parse(\'true\');'
+    CANNOT_EDIT_STR = 'GLOBALS.can_edit = JSON.parse(\'false\');'
+
+    def setUp(self):
+        """Completes the sign-up process for self.EDITOR_EMAIL."""
+        super(BaseCollectionEditorControllerTest, self).setUp()
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+
+        self.set_admins([self.ADMIN_EMAIL])
+
+    def assert_can_edit(self, response_body):
+        """Returns True if the response body indicates that the collection is
+        editable."""
+        self.assertIn(self.CAN_EDIT_STR, response_body)
+        self.assertNotIn(self.CANNOT_EDIT_STR, response_body)
+
+    def assert_cannot_edit(self, response_body):
+        """Returns True if the response body indicates that the collection is
+        not editable."""
+        self.assertIn(self.CANNOT_EDIT_STR, response_body)
+        self.assertNotIn(self.CAN_EDIT_STR, response_body)
+
+
+
+class CollectionEditorTest(BaseCollectionEditorControllerTest):
+    def setUp(self):
+        super(CollectionEditorTest, self).setUp()
+        collection_services.load_demo('0')
+        rights_manager.release_ownership_of_collection(
+            feconf.SYSTEM_COMMITTER_ID, '0')
+
+    def test_collection_editor_page(self):
+        """Test access to editor pages for the sample exploration."""
+
+        # Check that non-editors can access, but not edit, the editor page.
+        response = self.testapp.get('/collection_editor/create/0')
+        self.assertEqual(response.status_int, 200)
+        #self.assertIn('Welcome to Collections in Oppia!', response.body)
+        self.assert_cannot_edit(response.body)
+
+        # Log in as an editor.
+        self.login(self.EDITOR_EMAIL)
+
+        # Check that it is now possible to access and edit the editor page.
+        response = self.testapp.get('/collection_editor/create/0')
+        self.assertEqual(response.status_int, 200)
+        #self.assertIn('Welcome to Collections in Oppia!', response.body)
+        self.assert_can_edit(response.body)
+
```

Comment:
Since a redirect also happens when trying to edit a non-existent collection, that might be a way to test it (e.g. try to go to the editor page with a collection ID of 'invalid_col_id' or something similar should cause a redirect).